rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/services/ServiceConfig.java

```diff
@@ -42,6 +52,25 @@ Ok3Client okClient(OkHttp3ClientConfiguration okHttpClientConfig) {
     return new Ok3Client(okHttpClientConfig.create().build());
   }
 
+  /*OkHttpClient is creatig by calling {OkHttp3ClientConfiguration}.create() which has set all the properties.
+   * As part this OkHttpClient builder  OkHttp3MetricsInterceptor added as first interceptor, but this interceptor needs spinnaker
+   * headers but interceptor in okhttp are sequential, so altering the position of interceptor and resetting to client. */
+  @Bean
+  @Primary
```

Comment:
What happens if we remove this?  In general having to use `@Primary` is is a sign of problems.

Reply:
removed OkhttpClient bean created and moved the login as part ClouddriverRetrofit2Service bean